prov/shm/src/smr_attr.c

```diff
@@ -39,6 +39,7 @@
 struct fi_tx_attr smr_tx_attr = {
 	.caps = SMR_TX_CAPS,
 	.comp_order = FI_ORDER_STRICT,
+	.msg_order = FI_ORDER_STRICT,
```

Comment:
FI_ORDER_STRICT is a completion ordering semantic, not msg ordering semantic. msg_order should be stuff like SAS, SAW, WAR, etc. send/write/read after send/write/read.  msg_order will be restricted by the use of the fast RMA protocol.  IIUC, fast RMA results in reads or writes being seen out of order at the receiver.

Reply:
Actually, as it is right now I think fast RMA results would still be seen in order because the peer region is locked during the fast RMA so no one else can post to the receiver side until the CMA is over. Not optimal though. One of the areas I'm looking into is how to avoid this region locking during the memcpy but I think we can still keep it in order since the sender still needs a spot in the command queue and claims that before doing the copy.

Reply:
The problem is that fast RMA can be mixed with the non-fast RMAs.  The results is that a fast RMA can pass a 'slow' one which is sitting in the command queue.

Reply:
Ahh yes. Got it. So it won't be able to ensure any RMA ordering. Is this going to be an issue with compatibility?

Reply:
Well, since you dynamically enable fast RMA based on support for MR_VIRT_ADDR, you could add msg_order to that selection as well.  If the app requests S/R/W after R/W or R/W after S/R/W (basically anything except SAS ordering -- I think), you can support it by disabling fast RMA.